src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py

```diff
@@ -338,8 +350,7 @@ def __call__(
             mask_image (`PIL.Image.Image`):
                 `Image`, or tensor representing an image batch, to mask `image`. White pixels in the mask will be
                 repainted, while black pixels will be preserved. If `mask_image` is a PIL image, it will be converted
-                to a single channel (luminance) before use. If it's a tensor, it should contain one color channel (L)
-                instead of 3, so the expected shape would be `(B, H, W, 1)`.
+                to a single channel (luminance) before use. If it's a tensor, it should be of shape `(B, H, W)`.
```

Comment:
I'm not sure about this. The code in `_prepare_mask_and_masked_image` unconditionally prepends a batch dimension to the image. We also need to ensure that the mask has a single channel, so the shape in my opinion would have to be `(H, W, 1)` and then adapt the permutes accordingly.

We also need to document the expected shape of the `image` itself: `(H, W, 3)`, I guess?

Reply:
I was confused too by the docstrings. Before I even began work it said it supported batches and tensor inputs, but I couldn't get either to work.

I disagree about the mask input, I feel it would be cleaner to implement with it being shape `(H, W)`. But if there is some convention I am unaware of then I can change it.

I think for now I can update the doc strings for the single image case, then create a separate issue for multi batch. Would you agree? I have a local fork that adds batching to `_prepare_mask_and_masked_image`, but it fails further on in the code. It seems more involved and probably needs a separate issue.